File: nars_core_scala/src/main/scala/nars/gui/MessageDialog.scala
package nars.gui

import java.awt._
import java.awt.event._
//remove if not needed
import scala.collection.JavaConversions._

/**
 * Pop-up message for the user to accept
 */
class MessageDialog(parent: Frame, message: String) extends Dialog(parent, "Message", false) with ActionListener with WindowListener {

  protected var button: Button = new Button(" OK ")

  protected var text: TextArea = new TextArea(message)

//  super(parent, "Message", false)

  setLayout(new BorderLayout(5, 5))

  setBackground(NarsFrame.SINGLE_WINDOW_COLOR)

  setFont(NarsFrame.NarsFont)

  text.setBackground(NarsFrame.DISPLAY_BACKGROUND_COLOR)

  this.add("Center", text)

  button.addActionListener(this)

  val p = new Panel()

  p.setLayout(new FlowLayout(FlowLayout.CENTER, 5, 5))

  p.add(button)

  this.add("South", p)

  setModal(true)

  setBounds(200, 250, 400, 180)

  addWindowListener(this)

  setVisible(true)

  /**
   * Handling button click
   * @param e The ActionEvent
   */
  def actionPerformed(e: ActionEvent) {
    if (e.getSource == button) {
      close()
    }
  }

  private def close() {
    this.setVisible(false)
    this.dispose()
  }

  override def windowActivated(arg0: WindowEvent) {
  }

  override def windowClosed(arg0: WindowEvent) {
  }

  override def windowClosing(arg0: WindowEvent) {
    close()
  }

  override def windowDeactivated(arg0: WindowEvent) {
  }

  override def windowDeiconified(arg0: WindowEvent) {
  }

  override def windowIconified(arg0: WindowEvent) {
  }

  override def windowOpened(arg0: WindowEvent) {
  }
}

File: nars_gui/src/main/java/nars/gui/MessageDialog.java
/*
 * MessageDialog.java
 *
 * Copyright (C) 2008  Pei Wang
 *
 * This file is part of Open-NARS.
 *
 * Open-NARS is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 2 of the License, or
 * (at your option) any later version.
 *
 * Open-NARS is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Open-NARS.  If not, see <http://www.gnu.org/licenses/>.
 */
package nars.gui;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.awt.event.WindowEvent;
import java.awt.event.WindowListener;

/**
 * Pop-up message for the user to accept
 */
public class MessageDialog extends JDialog implements ActionListener, WindowListener {

    protected JButton button;
    protected JTextArea text;

    /**
     * Constructor
     * @param parent The parent Frame
     * @param message The text to be displayed
     */
    public MessageDialog(String message) {
        super((Dialog)null, "Message", false);
        setLayout(new BorderLayout(5, 5));
        
        text = new JTextArea(message);

        add("Center", text);
        button = new JButton(" OK ");
        button.addActionListener(this);
        JPanel p = new JPanel();
        p.setLayout(new FlowLayout(FlowLayout.CENTER, 5, 5));
        p.add(button);
        add("South", p);
        setModal(true);
        setBounds(200, 250, 400, 180);
        addWindowListener(this);
        setVisible(true);
    }

    /**
     * Handling button click
     * @param e The ActionEvent
     */
    @Override
    public void actionPerformed(ActionEvent e) {
        if (e.getSource() == button) {
            close();
        }
    }

    private void close() {
        setVisible(false);
        dispose();
    }

    @Override
    public void windowActivated(WindowEvent arg0) {
    }

    @Override
    public void windowClosed(WindowEvent arg0) {
    }

    @Override
    public void windowClosing(WindowEvent arg0) {
        close();
    }

    @Override
    public void windowDeactivated(WindowEvent arg0) {
    }

    @Override
    public void windowDeiconified(WindowEvent arg0) {
    }

    @Override
    public void windowIconified(WindowEvent arg0) {
    }

    @Override
    public void windowOpened(WindowEvent arg0) {
    }
}
